File: trip-kit-booking/src/main/java/com/skedgo/android/tripkit/booking/AccessTokenResponse.java
package com.skedgo.android.tripkit.booking;

import android.support.annotation.Nullable;

import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

@Value.Immutable
@Gson.TypeAdapters
@Value.Style(passAnnotations = JsonAdapter.class)
@JsonAdapter(AccessTokenResponse.class)
public interface AccessTokenResponse {

  @SerializedName("access_token") String accessToken();
  @SerializedName("token_type") String tokenType();
  @SerializedName("expires_in") long expiresIn();
  @Nullable @SerializedName("refresh_token") String refreshToken();

}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/AccessTokenResponse.kt
package com.skedgo.tripkit.booking

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    AccessTokenResponse::class
)
interface AccessTokenResponse {
    @SerializedName("access_token")
    fun accessToken(): String?

    @SerializedName("token_type")
    fun tokenType(): String?

    @SerializedName("expires_in")
    fun expiresIn(): Long

    @SerializedName("refresh_token")
    fun refreshToken(): String?
}
